src/components/CvssCalculator/CvssCalculatorBase.vue

```diff
@@ -0,0 +1,189 @@
+<script setup lang="ts">
+import { computed, onMounted, ref, toRaw, watch } from 'vue';
+
+import { equals } from 'ramda';
+
+import { useCvssScores, validateCvssVector } from '@/composables/useCvssScores';
+import { calculateCvss3Score, parseCvss3Factors } from '@/composables/useCvss3Calculator';
+
+import type { CvssEntity, ZodAffectCVSSType } from '@/types';
+import { CvssVersions } from '@/constants';
+import { deepCopyFromRaw } from '@/utils/helpers';
+
+import Cvss3Calculator from './Cvss3Calculator/Cvss3Calculator.vue';
+import CvssVectorInput from './CvssVectorInput.vue';
+
+const props = defineProps<{
+  cvssEntity: CvssEntity;
+}>();
+
+const emit = defineEmits<{
+  'blur': [FocusEvent];
+  'change:cvss_score': [ZodAffectCVSSType[]];
+}>();
+
+const highlightedFactor = ref<null | string>(null);
+const highlightedFactorValue = ref<null | string>(null);
+const cvssDiv = ref();
+const isFocused = ref(false);
+const cvssVectorInput = ref();
+const cvssEntity = ref<CvssEntity>(deepCopyFromRaw(toRaw(props.cvssEntity)));
+
+const {
+  cvss3Factors,
+  cvssScore,
+  cvssVector,
+  cvssVersion,
+  updateScore,
+  updateVector,
+} = useCvssScores(cvssEntity.value);
+
+const error = computed(() => validateCvssVector(cvssVector.value, cvssVersion.value) ?? null);
+
+watch(() => cvssEntity.value, () => {
+  if (!equals(props.cvssEntity.cvss_scores, cvssEntity.value.cvss_scores)) {
```

Comment:
```suggestion
watch(() => cvssEntity.value, (entity) => {
  if (!equals(props.cvssEntity.cvss_scores, entity.cvss_scores)) {
```

Are there cases where the above would be better?

Reply:
No specific reason, I like yours more